Implement an initial version of the support bundle in Alloy (#2009)
grafana/alloy: docs/sources/troubleshoot/support_bundle.md

---
canonical: https://grafana.com/docs/alloy/latest/troubleshoot/support_bundle/
description: Learn how to generate a support bundle
title: Generate a Support Bundle
menuTitle: Generate Support Bundle
weight: 300
---

> # Generate Support Bundle

Reviewer: Will this be released as Public preview?

If yes, then we probably should add the Public preview bits at the top of the doc topic.

```
<span class="badge docs-labels__stage docs-labels__item">Public preview</span>

# Generate Support Bundle

{{< docs/shared lookup="stability/public_preview.md" source="alloy" version="<ALLOY_VERSION>" >}}
```

Author: Should we use the shared component for public_preview? The language doesn't seem quite right for this feature

<img width="765" alt="image" src="https://github.com/user-attachments/assets/28b525bc-1ec9-4a7c-a836-229596d69d55">

Reviewer: Oooh good point. Lets switch to the generic one instead.  We could use this in its place: 

```
{{< docs/public-preview product="Generate support bundle" >
```

https://grafana.com/docs/writers-toolkit/write/shortcodes/#docspublic-preview

The endpoint is only exposed to the {{< param "PRODUCT_NAME" >}} HTTP server listen address, which
defaults to `localhost:12345`.

The support bundle contains all information in plain text, so you can
inspect it before sharing to verify that no sensitive information has leaked.

In addition, you can inspect the [supportbundle implementation](https://github.com/grafana/alloy/tree/internal/service/http/supportbundle.go)
to verify the code used to generate these bundles.

A support bundle contains the following data:
* `alloy-components.json` contains information about the [components][components] running on this {{< param "PRODUCT_NAME" >}} instance, generated by the 
`/api/v0/web/components` endpoint.
* `alloy-logs.txt` contains the logs during the bundle generation.
* `alloy-metadata.yaml` contains the {{< param "PRODUCT_NAME" >}} build version and the installation's operating system, architecture, and uptime.
* `alloy-metrics.txt` contains a snapshot of {{< param "PRODUCT_NAME" >}}'s internal metrics.
* `alloy-peers.json` contains information about the identified cluster peers of this {{< param "PRODUCT_NAME" >}} instance, generated by the 
`/api/v0/web/peers` endpoint.

> * `alloy-runtime-flags.txt` contains the values of the runtime flags available in {{< param "PRODUCT_NAME" >}}.

Author: There's several potential follow ups for added value, environment values, config file/remote cfg, I'd prefer to get this version merged and then we can revisit additional value.

---
canonical: https://grafana.com/docs/alloy/latest/troubleshoot/support_bundle/
description: Learn how to generate a support bundle
title: Generate a Support Bundle
menuTitle: Generate Support Bundle
weight: 300
---

# Generate Support Bundle

The `/-/support?duration=N` endpoint returns a 'support bundle', a zip file that contains information
about a running {{< param "PRODUCT_NAME" >}} instance, and can be used as a baseline of information when trying
to debug an issue.

{{< admonition type="note" >}}

> By default this endpoint is enabled, but may be disabled using the `--disable-support-bundle` runtime flag.

Reviewer: ```suggestion
This endpoint is enabled by default, but may be disabled using the `--disable-support-bundle` runtime flag.
```

{{< /admonition >}}

The duration parameter is optional, must be less than or equal to the
configured HTTP server write timeout, and if not provided, defaults to it.

> The endpoint is only exposed to {{< param "PRODUCT_NAME" >}}'s HTTP server listen address, which

Reviewer: ```suggestion
The endpoint is only exposed to the {{< param "PRODUCT_NAME" >}} HTTP server listen address, which
```

defaults to `localhost:12345`.

> The support bundle contains all information in plain text, so that it can be
> inspected before sharing, to verify that no sensitive information has leaked.

Reviewer: ```suggestion
The support bundle contains all information in plain text, so you can
inspect it before sharing to verify that no sensitive information has leaked.
```

In addition, you can inspect the [supportbundle implementation](https://github.com/grafana/alloy/tree/internal/service/http/supportbundle.go)

> to verify the code that is being used to generate these bundles.

Reviewer: ```suggestion
to verify the code used to generate these bundles.
```

A support bundle contains the following data:
* `alloy-components.json` contains information about the [components][components] running on this {{< param "PRODUCT_NAME" >}} instance, generated by the 
`/api/v0/web/components` endpoint.
* `alloy-logs.txt` contains the logs during the bundle generation.
* `alloy-metadata.yaml` contains the {{< param "PRODUCT_NAME" >}} build version and the installation's operating system, architecture, and uptime.
* `alloy-metrics.txt` contains a snapshot of {{< param "PRODUCT_NAME" >}}'s internal metrics.
* `alloy-peers.json` contains information about the identified cluster peers of this {{< param "PRODUCT_NAME" >}} instance, generated by the 
`/api/v0/web/peers` endpoint.

> * `alloy-runtime-config.txt` contains the values of the runtime flags available in {{< param "PRODUCT_NAME" >}}.

Reviewer 2: config is a bit misleading because it could be confused with the alloy config

Author: Sure, `runtime-flags` would probably be clearer.

* The `pprof/` directory contains Go runtime profiling data (CPU, heap, goroutine, mutex, block profiles) as exported by the pprof package. See the [profile][profile]
 documentation for more details on how to use this information.

> 

Reviewer 2: could you create a github issue for the live debugging support please?

---
canonical: https://grafana.com/docs/alloy/latest/troubleshoot/support_bundle/
description: Learn how to generate a support bundle

> title: Generate a Support Bundle

Reviewer: ```suggestion
title: Generate a support bundle
```

> menuTitle: Generate Support Bundle

Reviewer: ```suggestion
menuTitle: Generate a support bundle
```

weight: 300
---

<span class="badge docs-labels__stage docs-labels__item">Public preview</span>

> # Generate Support Bundle

Reviewer: ```suggestion
# Generate a support bundle
```

{{< docs/shared lookup="stability/public_preview.md" source="alloy" version="<ALLOY_VERSION>" >}}

> The `/-/support?duration=N` endpoint returns a 'support bundle', a zip file that contains information

Reviewer: ```suggestion
The `/-/support?duration=N` endpoint returns a support bundle, a zip file that contains information
```

This endpoint is enabled by default, but may be disabled using the `--disable-support-bundle` runtime flag.
{{< /admonition >}}

The duration parameter is optional, must be less than or equal to the
configured HTTP server write timeout, and if not provided, defaults to it.
The endpoint is only exposed to the {{< param "PRODUCT_NAME" >}} HTTP server listen address, which
defaults to `localhost:12345`.

The support bundle contains all information in plain text, so you can
inspect it before sharing to verify that no sensitive information has leaked.

In addition, you can inspect the [supportbundle implementation](https://github.com/grafana/alloy/tree/internal/service/http/supportbundle.go)
to verify the code used to generate these bundles.

A support bundle contains the following data:
* `alloy-components.json` contains information about the [components][components] running on this {{< param "PRODUCT_NAME" >}} instance, generated by the 
`/api/v0/web/components` endpoint.
* `alloy-logs.txt` contains the logs during the bundle generation.
* `alloy-metadata.yaml` contains the {{< param "PRODUCT_NAME" >}} build version and the installation's operating system, architecture, and uptime.

> * `alloy-metrics.txt` contains a snapshot of {{< param "PRODUCT_NAME" >}}'s internal metrics.

Reviewer: ```suggestion
* `alloy-metrics.txt` contains a snapshot of the internal metrics for {{< param "PRODUCT_NAME" >}}.
```

We should avoid using possessive product names

* `alloy-peers.json` contains information about the identified cluster peers of this {{< param "PRODUCT_NAME" >}} instance, generated by the 
`/api/v0/web/peers` endpoint.
* `alloy-runtime-flags.txt` contains the values of the runtime flags available in {{< param "PRODUCT_NAME" >}}.

> * The `pprof/` directory contains Go runtime profiling data (CPU, heap, goroutine, mutex, block profiles) as exported by the pprof package. See the [profile][profile]
>  documentation for more details on how to use this information.

Reviewer: ```suggestion
* The `pprof/` directory contains Go runtime profiling data (CPU, heap, goroutine, mutex, block profiles) as exported by the pprof package.
Refer to the [profile][profile] documentation for more details on how to use this information.
```
